src/components/application_manager/test/state_controller/state_controller_test.cc

```diff
@@ -2477,6 +2559,424 @@ TEST_F(StateControllerImplTest,
+      .WillOnce(Return(incorrect_app));
+  EXPECT_CALL(*simple_app_ptr_, app_id()).Times(0);
+  state_ctrl_->on_event(event);
+}
+
+TEST_F(StateControllerImplTest, OnEventOnAppActivated) {
+  using namespace constants;
+
+  smart_objects::SmartObject msg;
+  for (std::vector<am::ApplicationSharedPtr>::const_iterator it =
+           applications_list_.begin();
+       it != applications_list_.end();
+       ++it) {
+    uint32_t app_id = (*it)->app_id();
+    msg[am::strings::msg_params][am::strings::app_id] = app_id;
+    const hmi_apis::FunctionID::eType event_id =
+        hmi_apis::FunctionID::BasicCommunication_OnAppActivated;
+    am::event_engine::Event event(event_id);
+    event.set_smart_object(msg);
+
+    EXPECT_CALL(app_manager_mock_, application(app_id)).WillOnce(Return(*it));
+    // SetRegularState
+    EXPECT_CALL(*simple_app_ptr_, app_id()).WillRepeatedly(Return(app_id));
+    EXPECT_CALL(*simple_app_ptr_, IsAudioApplication())
+        .WillRepeatedly(Return(true));
+
+    smart_objects::SmartObjectSPtr activate_app =
+        utils::MakeShared<smart_objects::SmartObject>();
+    (*activate_app)[am::strings::params][am::strings::correlation_id] = kCorrID;
+    SetBCActivateAppRequestToHMI(hmi_apis::Common_HMILevel::FULL, kCorrID);
+    state_ctrl_->on_event(event);
+  }
+}
+
+TEST_F(StateControllerImplTest, IsStateActive) {
+  HmiStatePtr state = createHmiState(mobile_apis::HMILevel::HMI_FULL,
+                                     mobile_apis::AudioStreamingState::AUDIBLE,
+                                     mobile_apis::SystemContext::SYSCTXT_MAIN);
+  state->set_state_id(HmiState::STATE_ID_CURRENT);
+  EXPECT_TRUE(state_ctrl_->IsStateActive(state->state_id()));
+  state->set_state_id(HmiState::STATE_ID_REGULAR);
+  EXPECT_TRUE(state_ctrl_->IsStateActive(state->state_id()));
+  state->set_state_id(HmiState::STATE_ID_TTS_SESSION);
+  EXPECT_FALSE(state_ctrl_->IsStateActive(state->state_id()));
+}
+
+TEST_F(StateControllerImplTest, IsStateActiveApplyCorrectTempStates) {
+  InsertApplication(simple_app_);
+  ApplyTempStatesForApplication(*simple_app_ptr_, valid_state_ids_);
+  std::vector<am::HmiState::StateID>::const_iterator it =
+      valid_state_ids_.begin();
+  for (; it != valid_state_ids_.end(); ++it) {
+    EXPECT_TRUE(state_ctrl_->IsStateActive(*it));
+  }
+}
+
+TEST_F(StateControllerImplTest, IsStateActiveApplyNotCorrectTempStates) {
+  smart_objects::SmartObject msg;
+  msg[am::strings::msg_params][am::hmi_notification::event_name] =
+      hmi_apis::Common_EventTypes::AUDIO_SOURCE;
+  const hmi_apis::FunctionID::eType event_id = hmi_apis::FunctionID::VR_Started;
+  am::event_engine::Event event(event_id);
+  event.set_smart_object(msg);
+  state_ctrl_->on_event(event);
+  EXPECT_FALSE(state_ctrl_->IsStateActive(HmiState::STATE_ID_AUDIO_SOURCE));
+}
+
+TEST_F(StateControllerImplTest, OnApplicationRegisteredDifferentStates) {
+  const uint32_t app_id = simple_app_->app_id();
```

Comment:
@LitvinenkoIra , can you create const for app_id in separate namespace avoiding this calling almost in each test?